tests/views/schema.py

```diff
@@ -20,10 +21,121 @@ def has_permission(self, source: Any, info: Info[Any, Any], **kwargs: Any) -> bo
         return False
 
 
+class ConditionalFailPermission(BasePermission):
+    @property
+    def message(self):
+        return f"failed after sleep {self.sleep}"
+
+    async def has_permission(self, source, info, **kwargs: Any) -> bool:
+        self.sleep = kwargs.get("sleep", None)
+        self.fail = kwargs.get("fail", True)
+        if self.sleep is not None:
+            await asyncio.sleep(kwargs["sleep"])
+        return not self.fail
+
+
 class MyExtension(SchemaExtension):
```

Comment:
Can this use https://github.com/strawberry-graphql/strawberry/blob/b7982878f1b4ec73d74af4a86ab85fdeb921763b/tests/schema/extensions/test_extensions.py#LL194C7-L194C23 ?

Reply:
> It would be preferable to do it the other way round
sure